mulighetsrommet-api/src/test/kotlin/no/nav/mulighetsrommet/api/utbetaling/ArrangorflateServiceTest.kt

```diff
@@ -0,0 +1,173 @@
+        pdlClient = mockPdlClient(ArrangorflateTestUtils.createPdlMockEngine())
+        query = HentAdressebeskyttetPersonBolkPdlQuery(pdlClient)
+        arrangorflateService = ArrangorFlateService(query, database.db)
+    }
+
+    afterEach {
+        database.truncateAll()
+    }
+
+    afterSpec {
+        oauth.shutdown()
+    }
+
+    test("getUtbetalinger should return list of compact utbetalinger for arrangor") {
+        val result = arrangorflateService.getUtbetalinger(ArrangorflateTestUtils.underenhet.organisasjonsnummer)
+
+        result shouldHaveSize 2
+        result.any { it.id == utbetaling.id } shouldBe true
+        result.any { it.id == friUtbetaling.id } shouldBe true
+
+        val forsteUtbetaling = result.first { it.id == utbetaling.id }
+        forsteUtbetaling.belop shouldBe 10000
+        forsteUtbetaling.status shouldBe ArrFlateUtbetalingStatus.KLAR_FOR_GODKJENNING
+
+        val andreUtbetaling = result.first { it.id == friUtbetaling.id }
+        andreUtbetaling.belop shouldBe 5000
+        andreUtbetaling.status shouldBe ArrFlateUtbetalingStatus.KLAR_FOR_GODKJENNING
+    }
+
+    test("getUtbetaling should return utbetaling by ID") {
+        val result = arrangorflateService.getUtbetaling(utbetaling.id)
+
+        result.shouldNotBeNull()
+        result.id shouldBe utbetaling.id
+        result.beregning should beInstanceOf<UtbetalingBeregningForhandsgodkjent>()
+        (result.beregning as UtbetalingBeregningForhandsgodkjent).output.belop shouldBe 10000
+    }
+
+    test("getTilsagn should return arrangorflateTilsagn by ID") {
+        val result = arrangorflateService.getTilsagn(tilsagn.id)
+
+        result.shouldNotBeNull()
+        result.id shouldBe tilsagn.id
+        result.status.status shouldBe TilsagnStatus.GODKJENT
+    }
+
+    test("getTilsagnByOrgnr should return list of tilsagn for arrangor") {
+        val result = arrangorflateService.getTilsagnByOrgnr(ArrangorflateTestUtils.underenhet.organisasjonsnummer)
+
+        result shouldHaveSize 1
+        result[0].id shouldBe tilsagn.id
+        result[0].status.status shouldBe TilsagnStatus.GODKJENT
+    }
+
+    test("getArrangorflateTilsagnTilUtbetaling should return tilsagn for given gjennomforing and period") {
+        val periode = Periode(LocalDate.of(2024, 7, 1), LocalDate.of(2024, 8, 1))
+        val result = arrangorflateService.getArrangorflateTilsagnTilUtbetaling(GjennomforingFixtures.AFT1.id, periode)
+
+        result shouldHaveSize 1
+        result[0].id shouldBe tilsagn.id
+    }
+
+    test("getRelevanteForslag should return empty list when no forslag exists") {
+        val utbetalingDto = database.db.session {
+            queries.utbetaling.get(utbetaling.id)
+        }
+        utbetalingDto.shouldNotBeNull()
+
+        val result = arrangorflateService.getRelevanteForslag(utbetalingDto)
+
+        result shouldHaveSize 0
+    }
+
+    test("toArrFlateUtbetaling should convert a forhandsgodkjent utbetaling") {
```

Comment:
Det kunne kanskje være kjekt å ha to varianter av denne testen, den ene der vi gjør setup av `deltakerForslag`, slik som i denne testen og asserter at status er `VENTER_PA_ENDRING`, mens den andre ikke har noen `deltakerForslag` men da har status `KLAR_FOR_GODKJENNING`.

Alternativt kunne vi kanskje refaktorert `toArrFlateUtbetaling` ut av servicen og til å ikke være avhengig av selve databasen  (altså gjøre den til en pure function) men heller få all data inn som argumenter - da er det enda enklere å gjøre test-setup etc. siden all input til funksjonen styres direkte fra testen i stedet for via database- og servicelogikk. Det vil by på litt mer refaktoreringer, men kanskje det er verdt det. Kan ta en titt på det sammen om du har lyst :blush: